### PR TITLE
Vpc

### PR DESCRIPTION
**Pull request by atiknn**

### CODE REVIEW

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]
- Please move your files outside of the `learn-terraform-aws-instance` folder and put them into the root of the repository
- Please remove `aws_instance.app_server` we no longer need it

**Comment by atiknn** [comment]
> * Please move your files outside of the `learn-terraform-aws-instance` folder and put them into the root of the repository
> * Please remove `aws_instance.app_server` we no longer need it

Resolved

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,21 @@
+provider "aws" {
+     region  = "us-east-1"
+     access_key = "XXXX"
```
Comment (vladimirmukhin) [inline main.tf]: Hardcoding access keys inside code is bad practice even if you remove them when committing to git.
There is always a chance that you accidentally commit your access key and secret key or forget to delete them from code.

Pelase obtain your keys from environment variables or from a credential file instead

Any of these options will work well:
https://registry.terraform.io/providers/hashicorp/aws/latest/docs#environment-variables
https://registry.terraform.io/providers/hashicorp/aws/latest/docs#shared-configuration-and-credentials-files
https://registry.terraform.io/providers/hashicorp/aws/latest/docs#using-an-external-credentials-process

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      }
+    }
+
+    required_version = ">= 1.2.0"
+}
+
+
+
+
+
```
Comment (vladimirmukhin) [inline main.tf]: Please remove these blank lines

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]
Please remove `terraform.tfstate` and `terraform.tfstate.backup` from git

File: main.tf (hunk 2 above)
Comment (atiknn) [reply main.tf]: ok

**Comment by atiknn** [comment]
> `rm.tfstate` and `terraform.tfs`

Resolved

**Review by vladimirmukhin — APPROVED** [review]

**Pull request merged**